src/CommonProviderImplementation/Helpers.fs

```diff
@@ -171,9 +172,10 @@ module internal ProviderHelpers =
         using (logTime "Loading" sampleOrSampleUri) <| fun _ ->
     
         let tryGetResource() = 
```

Comment:
Lazy also should guarantee that there are no other thread at the same time running the same code with the same parameters. https://msdn.microsoft.com/en-us/library/dd997286(v=vs.110).aspx